src/main/java/com/mixpanel/android/mpmetrics/Tweaks.java

```diff
@@ -77,164 +41,204 @@
     public static final @TweakType int STRING_TYPE = 4;
 
     /**
-     * You can use this interface together with {@link #bind(String, Object, TweakChangeCallback)} to
-     * get callbacks when a new tweak value is received from Mixpanel.
-     */
-    public interface TweakChangeCallback {
-        /**
-         * onChange will be called at least once for each value of the tweak discovered.
-         * It will be called on the main UI thread. onChange may be called multiple times,
-         * both with values from Mixpanel or with the default tweak value.
-         */
-        void onChange(Object value);
-    }
-
-    /**
-     * Represents a tweak (but not a tweak value) known to the system. This class
+     * Represents the value and definition of a tweak known to the system. This class
      * is used internally to expose tweaks to the Mixpanel UI,
      * and will likely not be directly useful to code that imports the Mixpanel library.
      */
-    public static class TweakDescription {
-        public TweakDescription(@TweakType int aType, Object aDefaultValue, Number aMin, Number aMax) {
+    public static class TweakValue {
+        private TweakValue(@TweakType int aType, Object aDefaultValue, Number aMin, Number aMax, Object value) {
             type = aType;
             defaultValue = aDefaultValue;
             minimum = aMin;
             maximum = aMax;
+            this.value = value;
+        }
+
+        public TweakValue updateValue(Object newValue) {
+            return new TweakValue(type, defaultValue, minimum, maximum, newValue);
+        }
+
+        public String getStringValue() {
+            String ret = null;
+
+            try {
+                ret = (String) defaultValue;
+            } catch (ClassCastException e2) {
+                ; // ok
+            }
+
+            try {
+                ret = (String) value;
+            } catch (ClassCastException e) {
+                ; // ok
+            }
+
+            return ret;
+        }
+
+        public Number getNumberValue() {
+            Number ret = 0;
```

Comment:
if both of defaultValue and value are null, we are returning 0.. Is that expected? to me, this behavior is wrong since 0 is not acting like the default value.